### DISCUSSION TITLE
Album upload in the web ui from pc -> album incomplete

### DISCUSSION THREAD

**Discussion by 0verEngineer** [immich-app/immich#8831] [Q&A]
### The bug

I have a bunch of albums organized in directories, i upload them to immich with the "Create album" button, i noticed that some albums do not have all pictures so i used the "Add photos" button and selected the whole album again and this did not fix the problem, 10 photos are still missing.

I did following steps to check the issue:
1. I downloaded the incomplete album from immich
2. created a test user
3. uploaded the incomplete album as new album to the test user with the "Create album" button
4. I used the "Add photos" button in the album and selected all photos from the complete album and now it worked and the album in immich was complete

Because of this test steps i thought the problem is that all photos from the complete album i have locally are already in immich but in another album (or in no album) so they get detected as duplicates and are not re-uploaded (correct behavior), but they are also not added to the album (incorrect behavior)

But i cannot reproduce it in my test user, i did following to try it:
1. I deleted all stuff in the test user and also deleted everything from the trash
2. I created a test album and uploaded 2 photos from my album
3. I created another album and selected all photos to upload
4. I checked how many items are in the second album i created and it was complete

But on my other user this does not work, my album stays incomplete (missing 10 photos)

I consider this as a bug and i do not know how to proceed with the album upload now.

### The OS that Immich Server is running on

Docker

### Version of Immich Server

v1.100.0 

### Version of Immich Mobile App

irrelevant

### Platform with the issue

- [X] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
default from v1.100.0
```

### Your .env content

```Shell
# You can find documentation for all the supported env variables at https://immich.app/docs/install/environment-variables

# The location where your uploaded files are stored
UPLOAD_LOCATION=/mnt/nas/auto_backup/Pictures/Immich

# The Immich version to use. You can pin this to a specific version like "v1.71.0"
IMMICH_VERSION=release

# Connection secret for postgres. You should change it to a random password
DB_PASSWORD=...

# The values below this line do not need to be changed
###################################################################################
DB_HOSTNAME=immich_postgres
DB_USERNAME=postgres
DB_DATABASE_NAME=immich

REDIS_HOSTNAME=immich_redis
```

### Reproduction steps

```bash
Cannot reproduce it with my test user
```

**Comment by 0verEngineer** [comment]
I just found the problem, the 10 missing photos from that album were in the trash, i emptied the trash and re-uploaded all photos to that album and now it is complete.
Maybe immich should recover trashed items if they get re-uploaded?

**Comment by 0verEngineer** [comment]
Ok i have somehow still the same problem with another album, i uploaded it and there are about 20 photos missing in immich, how can this happen?
The upload dialog shows 740 items to process but only 724 photos are in the album after it is finished.

**Comment by mmomjian** [comment]
> Ok i have somehow still the same problem with another album, i uploaded it and there are about 20 photos missing in immich, how can this happen? The upload dialog shows 740 items to process but only 724 photos are in the album after it is finished.

The missing ones are most likely duplicates. What does the upload dialog show for these images? You can test with a new user.

As for the re-uploading of deleted items, this is not currently implemented.

If you have any other issues please feel free to include more details here as to how they can be reproduced.

**Comment by 0verEngineer** [comment]
The upload dialog is not readable because there are 2 toast notifications for every image in front of it...

With duplicates you mean duplicates in my album directory and not photos that are already in immich - because when they are already in immich they should get added to the album.

How does the immich duplicate detection work?

**Reply by mmomjian** [reply]
The notifications should go away once the upload is done and the dialog should be readable, in my experience.

Immich duplicate detection is by SHA-1 checksum.

I don't believe the duplicates get added to the album - I think there is a feature request for this but I don't find it right now.

**Comment by 0verEngineer** [comment]
The upload dialog disappears when the upload is finished, the toast notifications stay longer because there are just way to many of them.

I did a duplicate check with digiKam and it seems that they were duplicates, i will do some more checks in the evening.

Thank you

**Comment by mmomjian** [comment]
Makes sense. My guess would be duplicates. The notification dialog going away is annoying. Sorry about that.
